libc/src/__support/CPP/limits.h

```diff
@@ -24,74 +26,74 @@ constexpr unsigned long long ULLONG_MAX = ~0ULL;
 
 template <class T> class numeric_limits {
 public:
-  static constexpr T max();
-  static constexpr T min();
+  LIBC_INLINE static constexpr T max();
+  LIBC_INLINE static constexpr T min();
 };
 
 // TODO: Add numeric_limits specializations as needed for new types.
 
 template <> class numeric_limits<int> {
 public:
-  static constexpr int max() { return INT_MAX; }
-  static constexpr int min() { return INT_MIN; }
+  LIBC_INLINE static constexpr int max() { return INT_MAX; }
```

Comment:
This is already implied by the fact that the function is defined in the class declaration. @frobtech can we fix the clang -tidy to ignore this ?

Reply:
(also applies to pretty much all code in these files)

Reply:
This is a fundamental misunderstanding of the purpose of `LIBC_INLINE`.  It is never implicit, because it exists to be (sometimes) defined to something other than simply `inline`.

Reply:
I've read the documentation (libc/docs/code_style.rst), but it conflates `LIBC_INLINE` and `inline`:

```
Instead of using the ``inline`` keyword, they should be tagged with the   ``LIBC_INLINE``
```

(though it does mention implicitly inline functions):

```
 The ``LIBC_INLINE`` tag should also be added to functions which have
   definitions that are implicitly inline
```

I still kind of find the name misleading, maybe it should just avoid using `inline` which is already used for not-quite-the-same concept ?

Reply:
The meaning is "where you'd use the `inline` keyword or it would be implied by the language spec", so I'm not at a loss for what different name would be more clear.

Reply:
The `LIBC_INLINE` macro implies `inline`, but some of our users (specifically Fuchsia) redefine the macro themselves to ensure that inlined functions also have additional properties. In Fuchsia's case they use it to enforce internal linkage. Linked below is their definition, along with a comment explaining why this is necessary.

https://cs.opensource.google/fuchsia/fuchsia/+/main:zircon/system/ulib/c/string/mem-aliases-include/src/__support/macros/attributes.h;l=24

Reply:
Thanks for the explanations. Can we put these in the doc (libc/docs/code_style.rst) ? There are probably going to be other people like me who are going to wonder about that, and those latter explanations make a lot of sense.

Reply:
As of today, the clang-tidy makes sure that `LIBC_INLINE` is in first position. This makes sense given Fuchsia' s definition but this makes it impossible to use `LIBC_INLINE` functions with attributes: e.g.:
```
[[nodiscard]] LIBC_INLINE constexpr int countl_zero(T value) ; // clang-tidy unhappy
```
```
LIBC_INLINE [[nodiscard]] constexpr int countl_zero(T value) ; // clang unhappy
```